### PR TITLE
Fix source upload for split-workspace build job

### PR DESCRIPTION
**Pull request by AAlon**
Depends on https://github.com/aws-robotics/travis-scripts/pull/59 (see more context  there)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by mm318** [comment]
Can we make uploading sources part of the "Deploy" stage instead of the "Build & Bundle" stage?

**Comment by AAlon** [comment]
> Can we make uploading sources part of the "Deploy" stage instead of the "Build & Bundle" stage?

The reason I didn't move it into the deploy stage is because we're doing work to prepare the workspace as part of the build script, like installing `rosinstall` and invoking rosws: 

```
for WS in $WORKSPACES
do
  WS_DIR="/${ROS_DISTRO}_ws/src/${BUILD_DIR_NAME}/${WS}"
  echo "looking for ${WS}, $WS_DIR"
  if [ -d "${WS_DIR}" ]; then
    echo "WS ${WS_DIR} found, running rosws update"
    rosws update -t "${WS_DIR}"
  fi
done
```

So we'd need to add this which will complicate things a bit. I'm not against it but I think we still need the change at https://github.com/aws-robotics/travis-scripts/pull/59 in for backwards compatibility, at least until we migrate all repos to do it via a deploy stage (after which we should still have that logic in travis-scripts but flip the default behavior so that by default UPLOAD_SOURCES is false for build jobs). And the corresponding change here is very small so it'll be easy to undo.

Edit: this just made me realize that the solution in #59 is probably not good enough (only one workspace will have run `rosws update`)

**Added separate stage for source upload**

**Comment by AAlon** [comment]
Changed the branch of travis-scripts just for CI, will change it back before merging if everything's good.

File: .travis.yml [hunk 1]
```diff
@@ -46,11 +46,14 @@ deploy:
 jobs:
   include:
     - stage: Build & Bundle
-      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws"
+      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws" UPLOAD_SOURCES="false"
     - stage: Build & Bundle
-      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="simulation_ws"
+      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="simulation_ws" UPLOAD_SOURCES="false"
+    - stage: Upload sources
+      script: . .ros_ci/add_tag.sh && .ros_ci/prepare_sources_to_upload.sh && set +e
```
Comment (mm318) [inline .travis.yml:53]: Why is `. .ros_ci/add_tag.sh` still here?

File: .travis.yml [hunk 2]
```diff
@@ -46,11 +46,14 @@ deploy:
 jobs:
   include:
     - stage: Build & Bundle
-      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws"
+      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws" UPLOAD_SOURCES="false"
```
Comment (mm318) [inline .travis.yml:49]: Related to https://github.com/aws-robotics/travis-scripts/pull/60#discussion_r331276402, `UPLOAD_SOURCES` should just always be `false` in the context of `common_sa_build.sh`, so declaring it here is unnecessary.

File: .travis.yml (hunk 1 above)
Comment (AAlon) [reply .travis.yml:53]: It exports SA_VERSION which we need for `before_deploy` and `deploy`.

File: .travis.yml (hunk 2 above)
Comment (AAlon) [reply .travis.yml:49]: What do you mean? the default behavior is to upload sources as part of the build jobs, so I'm disabling it explicitly for this one.

File: .travis.yml (hunk 1 above)
Comment (mm318) [reply .travis.yml:53]: "Upload Sources" stage doesn't have `before_deploy` and `deploy`. Are you saying this export from "Upload Sources" stage needs to be propagated to the `before_deploy` and `deploy` of the "Deploy" stage?

File: .travis.yml (hunk 2 above)
Comment (mm318) [reply .travis.yml:49]: Never mind. `UPLOAD_SOURCES` is effectively `true` by default when undefined for backwards compatibility. Earlier I forgot about the need to be backwards compatible.

File: .travis.yml (hunk 1 above)
Comment (AAlon) [reply .travis.yml:53]: It doesn't have an override for those, so it will use the `before_deploy` and `deploy` steps that are defined globally.

**Review by mm318 — APPROVED** [review]

**Pull request merged**